pylint/lint/utils.py

```diff
@@ -5,14 +5,32 @@
 from __future__ import annotations
 
 import contextlib
+import os
 import sys
 import traceback
 from collections.abc import Iterator, Sequence
 from datetime import datetime
 from pathlib import Path
 
 from pylint.config import PYLINT_HOME
-from pylint.lint.expand_modules import get_python_path
+
+
+def get_python_path(filepath: str) -> str:
```

Comment:
Should we cache this and/or ``fix_import_path`` now that we launch the function a lot ?

Reply:
`fix_import_path` is an iterator that changes `sys.path` so I don't think we can cache that.

I'll cache `get_python_path`.

Reply:
Oh never mind, I don't think `get_python_path` ever gets called with the same argument, as it is part of the exploration of modules and we don't "explore the same module twice".